### PR TITLE
[KTX] Add port

### LINKED ISSUE #15113: [New Port Request] libktx
Library name: libktx

Library description: A small library of functions for writing and reading KTX files.

Source repository URL: https://github.com/KhronosGroup/KTX-Software/tree/master/lib

### PR DESCRIPTION
**Pull request by stephenjust**
Adds a port for libktx

- What does your PR fix? Fixes #15113

- Which triplets are supported/not supported? UWP not supported, x86-windows not supported
(see https://github.com/KhronosGroup/KTX-Software/issues/347#issuecomment-747246291)

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@stephenjust, thanks for your contribution!

Here is CI test failures with x64-windows/x86-windows/x64-windows-static/arrm64-windows.

```
CMake Error at D:/downloads/tools/cmake-3.18.4-windows/cmake-3.18.4-win32-x86/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find Bash (missing: BASH_EXECUTABLE)
Call Stack (most recent call first):
  D:/downloads/tools/cmake-3.18.4-windows/cmake-3.18.4-win32-x86/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/FindBash.cmake:95 (find_package_handle_standard_args)
  CMakeLists.txt:8 (find_package)

-- Configuring incomplete, errors occurred!
```

**Comment by stephenjust** [comment]
> @stephenjust, thanks for your contribution!
> 
> Here is CI test failures with x64-windows/x86-windows/x64-windows-static/arrm64-windows.
> ...

This project has a custom module to resolve Bash, I have some ideas for how I might override the behavior - still investigating.

File: ports/ktx/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "supports": "!(uwp | x86)",
+  "dependencies": [
+    "zstd"
+  ],
+  "features": {
+    "tools": {
+      "description": "Build tools"
+    },
+    "vulkan": {
+      "description": "Build Vulkan support"
```
Comment (PhoebeHui) [inline ports/ktx/vcpkg.json:16]: Have you test the 2 features locally?

File: ports/ktx/0001-Use-vcpkg-zstd.patch [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
++        find_library(ZSTD_LIBRARY NAMES zstdd zstd_staticd zstd zstd_static NAMES_PER_DIR)
++        find_path(ZSTD_H_PATH zstd.h REQUIRED)
+         target_link_libraries(
+             ${lib}
+-            ${CMAKE_CURRENT_SOURCE_DIR}/other_lib/win/Release-x64/zstd_static.lib
++            ${ZSTD_LIBRARY}
++        )
++        target_include_directories(
++            ${lib}
++            PRIVATE ${ZSTD_H_PATH}
```
Comment (PhoebeHui) [inline ports/ktx/0001-Use-vcpkg-zstd.patch:18]: When build the feature 'tools',  it failed with 
```
vcpkg\buildtrees\ktx\src\.0.0-beta5-c23f0e515e.clean\utils\scapp.h(7): fatal error C1083: Cannot open include file: 'zstd.h': No such file or directory
```
You may need to update the PRIVATE to PUBLIC.

BTW, you can also update the patch as following instead.
```
  find_package(zstd CONFIG REQUIRED)
  target_link_libraries(
            ${lib}
            $<IF:$<TARGET_EXISTS:zstd::libzstd_shared>,zstd::libzstd_shared,zstd::libzstd_static>
        )
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]

File: ports/ktx/0001-Use-vcpkg-zstd.patch (hunk 2 above)
Comment (stephenjust) [reply ports/ktx/0001-Use-vcpkg-zstd.patch]: Great suggestion here - seems to work and eliminates the need to manually include zstd.h.

File: ports/ktx/vcpkg.json (hunk 1 above)
Comment (stephenjust) [reply ports/ktx/vcpkg.json:16]: I've built the project with the different feature combinations, and I've run each of the tools on some sample files from the source repository. I unfortunately don't fully understand what the vulkan support is doing for any meaningful validation other than that the build output looks correct according to my interpretation of the cmake scripts.

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for your updates!

**Comment by vicroms** [comment]
Thanks for the PR!

**Pull request merged**